src/jit/flowgraph.cpp

```diff
@@ -8753,6 +8756,63 @@ BasicBlock* Compiler::fgSplitBlockAfterStatement(BasicBlock* curr, GenTree* stmt
 }
 
 //------------------------------------------------------------------------------
+// fgSplitBlockAfterNode - Split the given block, with all code after
+//                         the given node going into the second block.
+//                         This function is only used in LIR.
+//------------------------------------------------------------------------------
+BasicBlock* Compiler::fgSplitBlockAfterNode(BasicBlock* curr, GenTree* node)
+{
+    assert(curr->IsLIR());
+
+    BasicBlock* newBlock = fgSplitBlockAtEnd(curr);
+
+    if (node)
+    {
+        newBlock->bbTreeList = node->gtNext;
+        if (newBlock->bbTreeList != nullptr)
+        {
+            // Make gtPrev links circular
+            // REVIEW: necessary for LIR?
+            newBlock->bbTreeList->gtPrev = curr->bbTreeList->gtPrev;
+        }
+
+        // Make gtPrev links circular
+        curr->bbTreeList->gtPrev = node;
```

Comment:
This seems wrong: the LIR lists are not designed to be circular. Same for the mutation done above. I'd just remove these.